src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -38,55 +38,46 @@ public SecretMasker() : this(default, default, default, default, default)
     {
     }
 
-    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets,
+    public SecretMasker(IEnumerable<RegexPattern>? regexSecrets = null,
                         bool generateCorrelatingIds = false,
                         IRegexEngine? regexEngine = default,
                         string? defaultRegexRedactionToken = null,
                         string? defaultLiteralRedactionToken = null)
     {
-        _disposed = false;
-
-        RegexPatterns = regexSecrets != null
-            ? new HashSet<RegexPattern>(regexSecrets)
-            : new HashSet<RegexPattern>();
-
+        _regexPatterns = new HashSet<RegexPattern>(regexSecrets ?? []);
```

Comment:
>[] [](http://example.com/codeflow?start=67&length=2)

NIT: maybe simply set default arg to `[]` instead of null ?

Reply:
I wish that were possible but `[]` is not a compile-time constant and therefore can't be a default arg.